gnowsys-ndf/gnowsys_ndf/ndf/views/mio.py

```diff
@@ -0,0 +1,432 @@
+        msg['From'] = "ps.mio.bits@gmail.com"
+        msg['To'] = to_user
+        
+        context = {'pg_name': page_name,'pg_content': page_content}
+    	Html = render(None, 'ndf/mio_index.html', context)
+       	html = str(Html)
+       	html = html.strip('Content-Type: text/html; charset=utf-8\n')	
+        msg.attach(MIMEText(html, 'html'))
+        smtpObj=smtplib.SMTP('smtp.gmail.com', 587)
+        smtpObj.ehlo()
+        smtpObj.starttls()
+        smtpObj.login('ps.mio.bits@gmail.com','1Guesswhat')
+        smtpObj.sendmail('ps.mio.bits@gmail.com',to_user,msg.as_string())
+        smtpObj.quit()
+
+def update_page(name,content,id):
+	gst_page = node_collection.one({u'_id':ObjectId(id)})
+	gst_page.name = unicode(name)
+	gst_page.altnames = unicode(name)
+	gst_page.content = unicode(content)
+
+	try:
+		gst_page.save()
+	except:
+		print 'in except'
+
+def upload(gp_name,filename,name,author,content):
+	home_grp = node_collection.one({'_type': "Group", 'name':gp_name})
+	file_gst = node_collection.one({'_type': "GSystemType", 'name': u'File'})
+
+	each_img_name = filename
+	each_img_name_wo_ext = each_img_name.split('.')[0]
+	path = './attachment'
+
+	try:
+		file_obj_in_str = open(path+'/'+each_img_name,'r+')
+		#file_obj_in_str = open('./attachment/helloworld.txt','r+')
+	except Exception as e:
+		print "can't open" + str(e)
+
+	img_file = io.BytesIO(file_obj_in_str.read())
+	img_file.seek(0)
+	fh_obj = filehive_collection.collection.Filehive()
+	filehive_obj_exists = fh_obj.check_if_file_exists(img_file)
+	file_gs_obj = None
+
+	if not filehive_obj_exists:
+		file_gs_obj = node_collection.collection.GSystem()
+
+		file_gs_obj.fill_gstystem_values(
+										request=None,
+										name=unicode(name),
+										group_set=[home_grp._id],
+										# language=language,
+										uploaded_file=img_file,
+										created_by=author,
+										member_of=[file_gst._id],
+										origin={'user-icon-list-path':str(path)},
+										unique_gs_per_file=True,
+										content_org=content
+										)
+		print "\n\nGS obj id "+str(file_gs_obj._id)+"\n\n"
+		try:
+			# file_gs_obj.save(groupid=home_grp._id)
+			file_gs_obj.save(groupid=home_grp._id)
+		except Exception as e:
+			print 'exception while saving', e
+	else:
+		print "file exists"
+	return 'done'
+
+def authenticate_user(mail,group_name):
+	id = -1
+	try:
+		id = User.objects.get(email=mail).id
+	except:
+		return id,False,error_message["User does not exist"]
+	
+	# Extracting lists of all groups
+	gst_group = node_collection.find({'_type':u'Group'})
+	group_list = []
+	
+	for each in gst_group:
+		group_list.append(each.name)
+
+	if(group_name in group_list):
+		gst_home = get_group_name_id(group_name,get_obj=True)
+		if(id in gst_home.author_set):
+			return id,True,error_message["User authenticated"]
+		else:	
+			return id,False,error_message["User not a member of this group"]
+	else:
+		return id,False,error_message["Group does not exist"]
```

Comment:
You could try following for/replace/in above method:
<pre>
if node_collection.find({ '_type': 'Group', 'name': unicode(group_name), 'author_set': {'$in': [id]} }).count() > 0:
    return id,True,error_message["User authenticated"]
else:
    return id,False,error_message["User not a member of this group"]
</pre>